test/framework/reporter/esreporter.go

```diff
@@ -104,50 +99,63 @@ func NewDeprecatedGardenerESReporter(filename, index string) reporters.Deprecate
-}
+	reporter.testSuiteName = report.SuiteDescription
 
-// SpecDidComplete analysis the completed test and creates new es entry
-func (reporter *GardenerESReporter) SpecDidComplete(specSummary *types.SpecSummary) {
-	// do not report skipped tests
-	if specSummary.State == types.SpecStateSkipped || specSummary.State == types.SpecStatePending {
-		return
-	}
+	for _, spec := range report.SpecReports {
+		// do not report skipped tests
+		if spec.State == types.SpecStateSkipped || spec.State == types.SpecStatePending {
+			continue
+		}
 
-	testCase := TestCase{
-		Metadata:  reporter.suite,
-		Name:      strings.Join(specSummary.ComponentTexts[1:], " "),
-		ShortName: getShortName(specSummary.ComponentTexts[len(specSummary.ComponentTexts)-1]),
-		Phase:     PhaseForState(specSummary.State),
-		Labels:    parseLabels(strings.Join(specSummary.ComponentTexts[1:], " ")),
-	}
-	if specSummary.State == types.SpecStateFailed || specSummary.State == types.SpecStateInterrupted || specSummary.State == types.SpecStatePanicked {
-		testCase.FailureMessage = &FailureMessage{
-			Type:    PhaseForState(specSummary.State),
-			Message: failureMessage(specSummary.Failure),
+		var componentTexts []string
+		componentTexts = append(componentTexts, spec.ContainerHierarchyTexts...)
+		componentTexts = append(componentTexts, spec.LeafNodeText)
+		testCaseName := strings.Join(componentTexts[1:], " ")
+		testCase := TestCase{
+			Metadata:  reporter.suite,
+			Name:      testCaseName,
+			ShortName: getShortName(componentTexts[len(componentTexts)-1]),
+			Phase:     PhaseForState(spec.State),
+			Labels:    parseLabels(testCaseName),
 		}
-		testCase.SystemOut = specSummary.CapturedOutput
-		reporter.suite.Phase = SpecPhaseFailed
+
+		if spec.State == types.SpecStateFailed || spec.State == types.SpecStateInterrupted || spec.State == types.SpecStatePanicked {
+			reporter.suite.Failures++
+			testCase.FailureMessage = &FailureMessage{
+				Type:    PhaseForState(spec.State),
+				Message: failureMessage(spec.Failure),
+			}
+			testCase.SystemOut = spec.CombinedOutput()
+		}
+
+		testCase.Duration = spec.RunTime.Seconds()
+		reporter.testCases = append(reporter.testCases, testCase)
+
 	}
-	testCase.Duration = specSummary.RunTime.Seconds()
-	reporter.testCases = append(reporter.testCases, testCase)
-}
 
-// SuiteDidEnd collects the metadata for the whole test suite and writes the results
-// as elasticsearch json bulk to the specified location.
-func (reporter *GardenerESReporter) SuiteDidEnd(summary *types.SuiteSummary) {
-	reporter.suite.Tests = summary.NumberOfSpecsThatWillBeRun
-	reporter.suite.Duration = math.Trunc(summary.RunTime.Seconds()*1000) / 1000
-	reporter.suite.Failures = summary.NumberOfFailedSpecs
+	if reporter.suite.Failures != 0 {
+		reporter.suite.Phase = SpecPhaseFailed
+	}
+	reporter.suite.Tests = report.PreRunStats.SpecsThatWillRun
+	reporter.suite.Duration = math.Trunc(report.RunTime.Seconds()*1000) / 1000
 	reporter.suite.Errors = 0
```

Comment:
Question ->
Is there some reason we have the `Errors` field? It seems we are not utilizing it. And it was same in the past.

Reply:
True, we don't put anything meaningful into `Errors`. Unfortunately, I'm not too familiar with the history of the target es index, which structure I'm preserving here. 
@dguendisch do you happen to recall anything about this field?
Otherwise, I'd suggest we keep it for the time being.

Reply:
I think the basic semantic is that `Failures` captures "real" test failures and `Errors` captures situations where the test errors out (not necessarily because a broken assertion but because of an unexpected error).
Hence I think that line 131 should be changed to increment `Failures` only in `SpecStateFailed` and increment `Errors` in the two other cases.
In any case it should only have effects on our dashboards, but I think the above proposal would use the fields in a better way.